var/spack/repos/builtin/packages/ghost/package.py

```diff
@@ -0,0 +1,80 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Ghost(CMakePackage):
+    """GHOST: a General, Hybrid and Optimized Sparse Toolkit
+    
+       This library provides highly optimized building blocks for implementing sparse iterative eigenvalue and linear solvers
+       multi- and manycore clusters and on heterogenous CPU/GPU machines. For an iterative solver library using these kernels, see 
+       the phist package.
+    
+    """
+
+    homepage = "https://www.bitbucket.org/essex/ghost/"
+    url      = "https://bitbucket.org/essex/ghost/"
+
+    version('develop', git='https://bitbucket.org/essex/ghost/ghost.git', branch='devel')
+                                
+
+    def cmake_args(self):
+        spec=self.spec
+        cblas_include_dir=''
+        if '~mkl' in spec:
+            cblas_include_dir='-DCBLAS_INCLUDE_DIR='+spec['blas'].prefix.include
+
+        args = ['-DGHOST_ENABLE_MPI:BOOL=%s' % ('ON' if '+mpi' in spec else 'OFF'),
+                '-DGHOST_USE_CUDA:BOOL=%s'   % ('ON' if '+cuda' in spec else 'OFF'),
+                '-DGHOST_USE_SCOTCH:BOOL=%s'   % ('ON' if '+scotch' in spec else 'OFF'),
+                '-DGHOST_USE_ZOLTAN:BOOL=%s'   % ('ON' if '+zoltan' in spec else 'OFF'),
+                '-DBUILD_SHARED_LIBS:BOOL=%s' % ('ON' if '+shared' in spec else 'OFF'),
+                cblas_include_dir,
+                ];
+
+        return args
+
+    def check(self):
+        make('test')
+
+    variant('shared',  default=True, description='Enables the build of shared libraries')
+    variant('mpi', default=True, description='enable/disable MPI')
+    variant('cuda',default=False,description='enable/disable CUDA')
+    variant('mkl',default=False,description='use MKL to provide BLAS routines')
```

Comment:
i would not add this variant, you can figure out wherther `mkl` was chosen after the spec is conretized.

Reply:
see the (ugly) [numpy](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/py-numpy/package.py#L108-L139) example.

Reply:
things get uggly as soon as one depends on cblas (or lapacke), because there is no virtual package for those and the different vendors use different library names and may or may not include cblas/lapacke in the package providing blas/lapack. It would seem best to fix this elsewhere instead of me putting lots of if Statements in the package file (copied from the ugly Mumps example, ore or less)

Reply:
You don’t need virtual package for those. I think I commented on another PR that netlib-lapack can already return lapacke via this query mechanism. So IMO these things are best handled in lapack/blas providers.